modules/widgets/simple-payments/customizer.js

```diff
@@ -0,0 +1,294 @@
+/* global jQuery, jpSimplePaymentsStrings */
+/* eslint no-var: 0, quote-props: 0 */
+
+( function( api, wp, $ ) {
+	var $document = $( document );
+
+	$document.ready( function() {
+		$document.on( 'widget-added', function( event, widgetContainer ) {
+			if ( widgetContainer.is( '[id*="jetpack_simple_payments_widget"]' ) ) {
+				initWidget( widgetContainer );
+			}
+		} );
+
+		$document.on( 'widget-synced widget-updated', function( event, widgetContainer ) {
+			//this fires for all widgets, this prevent errors for non SP widgets
+			if ( ! widgetContainer.is( '[id*="jetpack_simple_payments_widget"]' ) ) {
+				return;
+			}
+
+			event.preventDefault();
+			var widgetForm = widgetContainer.find( '> .widget-inside > .form, > .widget-inside > form' );
+
+			if ( ! widgetForm.find( '.jetpack-simple-payments-form' ).is( ':visible' ) ) {
+				widgetForm.find( '.jetpack-simple-payments-add-product' )
+					.add( '.jetpack-simple-payments-edit-product' )
+					.add( '.jetpack-simple-payments-products' )
+					.removeAttr( 'disabled' );
+			} else {
+				widgetForm.find( '.jetpack-simple-payments-save-product' )
+					.add( '.jetpack-simple-payments-cancel-form' )
+					.add( '.jetpack-simple-payments-delete-product' )
+					.removeAttr( 'disabled' );
+			}
+
+			var newImageId = parseInt( widgetForm.find( '.jetpack-simple-payments-form-image-id' ).val(), 10 );
+			var newImageSrc = widgetForm.find( '.jetpack-simple-payments-form-image-src' ).val();
+
+			var placeholder = widgetForm.find( '.jetpack-simple-payments-image-fieldset .placeholder' );
+			var image = widgetForm.find( '.jetpack-simple-payments-image > img' );
+			var imageControls = widgetForm.find( '.jetpack-simple-payments-image' );
+
+			if ( newImageId && newImageSrc ) {
+				image.attr( 'src', newImageSrc );
+				placeholder.hide();
+				imageControls.show();
+			} else {
+				placeholder.show();
+				image.removeAttr( 'src' );
+				imageControls.hide();
+			}
+		} );
+	} );
+
+	function initWidget( widgetContainer ) {
+		var widgetForm = widgetContainer.find( '> .widget-inside > .form, > .widget-inside > form' );
```

Comment:
No need for such a large refactor, but this may be a better fit for a delegated event handler, for example if we have a consistent parent such as:
```
$('.wpbody-content').on( 'click', '.jetpack-simple-payments-add-product', showAddNewForm );
```
This would allow us to only need to bind events once, but we'd need to rewrite handlers to be able to find the relevant widget to react to the correct form. Usually it's enough to do actions relative the to the closest parent SP widget.

http://api.jquery.com/on/